qcodes/utils/update/qcodes_updater.py

```diff
@@ -0,0 +1,127 @@
+import os
+import subprocess
+import inspect
+import shutil
+import datetime
+import warnings
+import git
+
+import qcodes
+
+def qcodes_backup(env_name: str = 'qcodes', env_backup_name: str = 'qcodes_backup'):
+    """
+    Back up QCoDeS Anaconda Python environment.
+
+    Args:
+        env_name: Name of the environment to be updated
+        env_backup_name: Name of the back up environment to be created
+
+    Raises:
+        git.exc.GitError: If working tree is dirty
+    """
+    # QCoDeS source path
+    _source = os.sep.join(qcodes.__file__.split(os.sep)[:-2])
+    # Make sure that git working tree is clean
+    repo = git.Repo(_source)
+    if (inspect.stack()[1].function == 'update_qcodes_installation' and repo.is_dirty()):
+        raise git.exc.GitError
+    elif (inspect.stack()[1].function == 'conda_env_update' and repo.is_dirty()):
+        raise git.exc.GitError
+    elif repo.is_dirty():
+        print("QCoDeS working tree is not clean. Please commit or stash your changes, and try again.")
+    else:
+        _time = str(datetime.date.today()) # Current date.
+        env_backup_name = env_backup_name + '_' + _time
+
+        print(f"Existing {env_name} environment will be backed up as {env_backup_name}...\n")
+        subprocess.run(f'conda create --name {env_backup_name} --clone {env_name}', shell=True)
+
+        # Copy QCoDeS root to qcodes_backup
+        _destination = _source + '_backup' + '_' + _time
+        shutil.copytree(_source, _destination, symlinks=True, ignore=None)
+
+        # Re-install QCoDeS from the back up root
+        subprocess.run(f'activate {env_backup_name} && pip uninstall qcodes && pip install -e {_destination}', shell=True)
+
+        return _source, _destination, env_name, env_backup_name
```

Comment:
In general there is no need to use private `_` variables inside a function. They are private to the function scope anyway